src/Rings/msolve/f4.jl

```diff
@@ -105,7 +105,7 @@ function f4(
           cglobal(:jl_free), gb_len, gb_exp, gb_cf, jl_ld, field_char)
 
     vars = gens(R)[eliminate+1:end]
-    I.gb[degrevlex(vars)] = BiPolyArray(basis, keep_ordering=false, isGB=true)
+    I.gb[degrevlex(vars)] = IdealGens(basis, keep_ordering=false, isGB=true)
     
     return basis
```

Comment:
In order to coincide with the return value of `std_basis` and `groebner_basis` `f4` should also return the `IdealGens` structure of the GB:
```suggestion
    return I.gb[degrevlex(vars)]
```